riscv/trace_encoder_n.cc

```diff
@@ -0,0 +1,211 @@
+#include "trace_encoder_n.h"
+
+trace_encoder_n::trace_encoder_n() {
+  this->trace_sink= fopen("trace_n.bin", "wb");
```

Comment:
This file name should be different for each hartId .. a simple way to avoid clobbering the same file in a multi-core with-trace simulation.